### DISCUSSION TITLE
RFC: Relative Imports vs Absolute Imports

### DISCUSSION THREAD

**Discussion by beraldoleal** [avocado-framework/avocado#3525] [RFC]
Hi all,

I noticed that we have some selftests that checks for absolute imports on core (`selftests/modules-boundaries.sh`).

From PEP8:

> Absolute imports are recommended, as they are usually more readable and tend to be better behaved (or at least give better error messages) if the import system is incorrectly configured (such as when a directory inside a package ends up on sys.path)...

> However, explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose.

> Standard library code should avoid complex package layouts and always use absolute imports.

I just would like to know what you think about that. What do you prefer and why?

Does anyone have any strong objections to removing relative imports?

**Comment by willianrampazzo** [comment]
My preference is always for absolute imports, as they make the code clear to read.

I understand that relative imports can help in the development process while setting up a development environment, but, if possible, I would go with absolute imports.

In case it is not a pain and do not breaks what we have now, I agree with removing the relative imports.

**Comment by apahim** [comment]
+1 for absolute imports between sub-packages, nor +1 or -1 for absolute imports between modules of the same sub-package, as the use cases for relative imports there are stronger.

**Comment by beraldoleal** [comment]
@apahim can you elaborate on the "relative imports there are stronger", please?

**Comment by apahim** [comment]
https://www.python.org/dev/peps/pep-0328/#rationale-for-relative-imports

**Comment by apahim** [comment]
But as I said, nor +1 or -1. If you change it, awesome, if you don't, fine :)

**Comment by beraldoleal** [comment]
Haa ok, thanks!

**Comment by beraldoleal** [comment]
@clebergnu are you ok with moving from relative to absolute ? I would like to remove the `module-boundaries.sh` at least for `avocado/core`. What do you think?

**Reply by clebergnu** [reply]
I'm fine with setting on absolute only.  Just make sure that we don't regress wrt other stuff using `avocado/core`.
